### ISSUE TITLE
OBL_UNSATISFIED_OBLIGATION false positive with Java 9

### ISSUE THREAD

**Issue by ghenkes** [spotbugs/spotbugs#432]
After uplifting to compile using Java 9 the following code is being flagged for an OBL_UNSATISFIED_OBLIGATION. The same code passed inspection using Java 8. This may be related to #293.

```java
import java.sql.Connection;
import java.sql.PreparedStatement;
import java.sql.SQLException;

import org.apache.derby.jdbc.EmbeddedDataSource;

class TestResource {

    private static final String CREATE_TABLE_SQL_FORMAT = "create table %s (id varchar(100) not null) ";

    void create(final String tableName) throws SQLException {
        final EmbeddedDataSource ds = new EmbeddedDataSource();
        ds.setDatabaseName("derbydb");
        try (Connection connection = ds.getConnection();
                PreparedStatement statement = connection
                               .prepareStatement(String.format(CREATE_TABLE_SQL_FORMAT, tableName))) {
            statement.executeUpdate();
        }
    }
}
```

```
[INFO] TestResource.create(String) may fail to clean up java.sql.Statement [TestResource, TestResource, TestResource, TestResource, cTestResource] Obligation to clean up resource created at TestResource.java:[line 16] is not dischargedPath continues at TestResource.java:[line 14]Path continues at TestResource.java:[line 17]Path continues at TestResource.java:[line 18]Path continues atTestResource.java:[line 19] OBL_UNSATISFIED_OBLIGATION
```

**Comment by eolivelli** [comment]
I am upgrading from java8 to java10 and I am falling into this issue.

**Comment by KengoTODA** [comment]
#493 explains detail about this problem. I guess we need breaking change to support this case.

**Comment by ptahchiev** [comment]
Same here.

**Comment by eolivelli** [comment]
News on this issue ?

**Comment by KengoTODA** [comment]
No news we have. Your contribution is welcome!

**Comment by eolivelli** [comment]
@KengoTODA 
Do you have any hint about where to start?

**Comment by KengoTODA** [comment]
Please check https://github.com/spotbugs/spotbugs/issues/493
